core/operation/src/main/java/uk/gov/gchq/gaffer/jobtracker/JobTracker.java

```diff
@@ -33,6 +34,8 @@
 public class JobTracker extends Cache<String, JobDetail> {
 
     private static final String CACHE_SERVICE_NAME_PREFIX = "JobTracker";
+    // Executor to queue to allow queuing up async operations on the cache
+    private final ThreadPoolExecutor executor = new ThreadPoolExecutor(1, 1, 1, TimeUnit.MINUTES, new LinkedBlockingQueue<>());
```

Comment:
I assume this was done to improve performance, but there's nothing to prevent cache race conditions (which would be hard to test for). As such I'd suggest raising this as a separate ticket for a future improvement.
The PR description states only gets are async?

Reply:
So the race condition should be mitigated as only additions to the job tracker cache are async and link queued up to preserve order. Gets from cache are synced as if requested it will wait on the queue to finish so any additions in the executor queue are completed first, this is done via the `get()` on the executor submit.

Reply:
I see; am I right in thinking that the benefit from this is that calls to these methods will now return right away?
I'd suggest expanding the comment a little to explain the motivation and make clear there is no parallel execution.